### DISCUSSION TITLE
[@types/quill] Property 'history' does not exist on type 'Quill'.

### DISCUSSION THREAD

**Discussion by invissiblecat** [DefinitelyTyped/DefinitelyTyped#65936] [Issues with a @types package]
My stack: 

    "ngx-quill": "^16.2.1",
    "quill": "^1.3.7",
    "@types/quill": "^1.3.7"

I implementing undo and redo functionality using `.history.undo()`:

```
  @ViewChild("textEditor") textEditor: QuillEditorComponent;

  modules: QuillModules = { toolbar: {
    container: [["undo", "redo"]],
    handlers: {
      'undo': () => this.textEditor.quillEditor.history.undo(),
      'redo': () => this.textEditor.quillEditor.history.redo(),
    },
   }
}
```

Everything is working perfectly, only type give the error `Property 'history' does not exist on type 'Quill'.`. Upgrading to  `"@types/quill": "^2.0.1"` does help, but then comes the `Module '"quill"' has no exported member 'Delta'.` error  which is solved by downgrading to `"@types/quill": "^1.3.7"`. Here comes the infinite loop...

[a comment by an automated account is not shown]

**Comment by 43081j** [comment]
i'm not 100% sure (need to catch up on quill again) but i seem to remember quill publishing a 2.x years ago that never left beta

so the types in DT are for 2.x and the version in wide usage is 1.x

makes it awkward for us to fix this stuff i suspect, as it may only make sense in 1.x...
